datastore/longhorn.go

```diff
@@ -1723,7 +1723,7 @@ func (s *DataStore) CheckDataEngineImageCompatiblityByImage(image string, dataEn
 		return err
 	}
 
-	if engineImage.Status.Incompatible {
+	if engineImage.Status.Incompatible && volume.Status.State != longhorn.VolumeStateAttached {
```

Comment:
Why do you check whether the volume is attached?

Reply:
Because the volume can be attached, we should allow the rebuilding or expanding actions.

>As discussed, I think what you need to do is just remove validation from the engine & replica, as we only need to prevent the following two cases.
>- new volumes with incompatible image should not be created
>- existing volumes updated with incompatible image should not be updated.

Yes, I will update the commit as we discussed.

Reply:
> Because the volume can be attached, we should allow the rebuilding or expanding actions.

This is more like a workaround .. it would make `CheckDataEngineImageCompatiblityByImage` not work as expected.